src/Client/Donation.php

```diff
@@ -123,23 +70,30 @@ public function put(DonationModel $donation): bool
                 throw new NotFoundException();
             }
 
-            // All other errors should be logged so we get a notification and the app left to retry the
-            // push at a later date.
-            $this->logger->warning(sprintf(
-                'Donation update exception %s: %s. Body: %s',
+            $this->logger->error(sprintf(
+                'Donation upsert exception for donation UUID %s %s: %s. Body: %s',
+                $donation->getUuid(),
                 get_class($ex),
                 $ex->getMessage(),
                 $ex->getResponse() ? $ex->getResponse()->getBody() : 'N/A',
             ));
 
-            return false;
+            throw new BadRequestException('Donation not created');
         }
 
-        return ($response->getStatusCode() === 200);
-    }
+        if ($response->getStatusCode() > 201 || $response->getStatusCode() < 200) {
```

Comment:
this might be clearer?

```suggestion
        if (! in_array($response->getStatusCode(), [200, 201], true)) {

```